### DISCUSSION TITLE
ctl embbeded CtrlcppCheck empty output Invalid mode: 'rU'

### DISCUSSION THREAD

**Discussion by cbv08** [siemens/CtrlppCheck#73] [Show and tell]
This is not excactly a bug report, but i thought it would be good to capture it somewhere. So since we have this nice new forum I elected to put it here.

I installed CtrlppCheck v1.0.x, Python 3.12 and Lizard. Whenever i openend the staticTest in a ctl script the ouput would look something like this:
![image](https://github.com/siemens/CtrlppCheck/assets/147402639/9186bac5-4d38-4ca2-8c01-e757e874fcb8)

And there would be nothing in the logs. To get the log output use the following arguement when calling GEDI
`-dbg ScriptData`
To get to this point took me a while.
The error i got was essentially: ValueError: invalid mode: 'rU'
Which appears because that read mode was deprecated in Python 3.1 and completely removed in Python 3.8
I used mode 'r' instead, and it works flawlessly
![image](https://github.com/siemens/CtrlppCheck/assets/147402639/dbfad9c7-4d16-4f7b-929a-7428747d0952)

This might be obvious to most but as no errors was thrown in the WinCC OA log it took me a while to find the issue.

If I should instead make a bug report, please write it below. I'll happily do that too.
Kind regards
Olav

**Comment by dhoegerlETM** [comment]
Hello Olav! With correct setup and configuration of CtrlppCheck subproject I cannot see your mentioned error from Python script. Please report in more detail and yes, if you find a problem in our documentation or tool usage please report an issue accordingly.

Best, Daniel

**Reply by cbv08** [reply]
Hello Daniel! Which version of python are you running? As stated in my post I installed version 3.12. Which, granted, is significantly higher version compared to the "minimum version 3.6" stated in the user documentation. 

However as far as i can tell the open mode: 'rU' was deprecated back in version 3.1 of python, and was completely removed in version 3.8. So I guess if you are running a version of python older than version 3.8 then you would not see this problem.

If you are running a newer version of Python, then I have no idea why it did not work for me though

Kind regards
Olav

**Reply by dhoegerlETM** [reply]
I'm running Python 3.9.6 so it might well be an issue when using newer versions, so please report an issue with more details about the message and from what script it originates.

Thanks, Daniel

**Reply by cbv08** [reply]
Ah sorry, I just found the python documentation for open: 
Changed in version 3.11: The 'U' mode has been removed.
cite: https://docs.python.org/3/library/functions.html#open

3.8 was from older documentation, I have created an issue.
Kind regards 
Olav
